ReactNativeClient/lib/components/screens/note.js

```diff
@@ -291,6 +293,8 @@ class NoteScreenComponent extends BaseScreenComponent {
 	}
 
 	async UNSAFE_componentWillMount() {
+		Object.assign({}, this.selection);
```

Comment:
This looks accidental.

Reply:
Not really - this note component is created once and then mounted/unmounted to show different notes. I'm invalidating selection to make sure Joplin doesn't try to insert at the selection from the previous note which might not be valid for the current one.

But I should move it to componentDidMount as the usage of this method is discouraged -- will do now.

Reply:
Could you explain how this code invalidates this.selection? Looks like a NOP to me

Reply:
Ah i see what you mean now. I'll fix it.

Reply:
done

Reply:
Object.assign only copies the key,value pairs from the right hand argument to the left. It does not overwrite. This line is still a NOP

Reply:
Thanks. Fixed now. I think.

Reply:
Looks good!